ServerTaskEngineBlockingTest is failing with JDK17

### Describe the bug
Two tests are failing with JDK17 in relation to virtual threads:
* `org.infinispan.server.task.ServerTaskEngineBlockingTest.testBlockingTaskOnBlockingManager`
* `org.infinispan.jmx.ClusteredCacheManagerMBeanTest.testExecutorMBeans`
  * expected:<-1> but was:<30000>

```
java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.AssertionError: Thread[blocking-thread-ServerTaskEngineBlockingTest-NodeA-p8-t2,5,ISPN-blocking-group] is not a virtual thread
	at org.infinispan.test.TestingUtil.withCacheManager(TestingUtil.java:1563)
	at org.infinispan.server.task.ServerTaskEngineBlockingTest.testBlockingTaskOnBlockingManager(ServerTaskEngineBlockingTest.java:34)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:61)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.apache.maven.surefire.junitcore.pc.Scheduler$1.run(Scheduler.java:345)
	at org.apache.maven.surefire.junitcore.pc.InvokerStrategy.schedule(InvokerStrategy.java:47)
	at org.apache.maven.surefire.junitcore.pc.Scheduler.schedule(Scheduler.java:316)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.infinispan.commons.test.junit.JUnitThreadTrackerRule$1.evaluate(JUnitThreadTrackerRule.java:26)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.apache.maven.surefire.junitcore.pc.Scheduler$1.run(Scheduler.java:345)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.util.concurrent.ExecutionException: java.lang.AssertionError: Thread[blocking-thread-ServerTaskEngineBlockingTest-NodeA-p8-t2,5,ISPN-blocking-group] is not a virtual thread
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
	at org.infinispan.server.task.ServerTaskEngineBlockingTest$1.call(ServerTaskEngineBlockingTest.java:76)
	at org.infinispan.test.TestingUtil.withCacheManager(TestingUtil.java:1559)
	... 34 more
Caused by: java.lang.AssertionError: Thread[blocking-thread-ServerTaskEngineBlockingTest-NodeA-p8-t2,5,ISPN-blocking-group] is not a virtual thread
	at org.infinispan.server.task.ServerTaskEngineBlockingTest$1$1.call(ServerTaskEngineBlockingTest.java:46)
	at org.infinispan.server.task.ServerTaskEngineBlockingTest$1$1.call(ServerTaskEngineBlockingTest.java:39)
	at org.infinispan.server.tasks.ServerTaskWrapper.run(ServerTaskWrapper.java:44)
	at org.infinispan.server.tasks.ServerTaskWrapper.apply(ServerTaskWrapper.java:50)
	at org.infinispan.server.tasks.ServerTaskWrapper.apply(ServerTaskWrapper.java:20)
	at org.infinispan.security.Security.doAs(Security.java:114)
	at org.infinispan.server.tasks.LocalServerTaskRunner.lambda$execute$0(LocalServerTaskRunner.java:28)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1586)
	... 1 more
```

### Expected behavior
Test passes with JDK17

### Actual behavior
Test fails with JDK17

### How to Reproduce?
Set JAVA_HOME to JDK17 and run the test

### Output of `uname -a` or `ver`
N/A

### Output of `java -version`
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8
openjdk version "17.0.12" 2024-07-16 LTS
OpenJDK Runtime Environment (Red_Hat-17.0.12.0.7-1) (build 17.0.12+7-LTS)
OpenJDK 64-Bit Server VM (Red_Hat-17.0.12.0.7-1) (build 17.0.12+7-LTS, mixed mode, sharing)

### Infinispan version or git rev
N/A

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)
N/A

### Additional information
N/A

--- a/commons/all/src/main/java/org/infinispan/commons/jdkspecific/ThreadCreator.java
+++ b/commons/all/src/main/java/org/infinispan/commons/jdkspecific/ThreadCreator.java
@@ -16,7 +16,7 @@ public class ThreadCreator {
    private static org.infinispan.commons.spi.ThreadCreator INSTANCE = getInstance(useVirtualThreads);
 
    public static void useVirtualThreads(boolean useVirtualThreads) {
-      if (useVirtualThreads != ThreadCreator.useVirtualThreads) {
+      if (useVirtualThreads != useVirtualThreads()) {
          ThreadCreator.useVirtualThreads = useVirtualThreads;
          INSTANCE = getInstance(useVirtualThreads);
       }
@@ -24,6 +24,10 @@ public class ThreadCreator {
 
    public static boolean useVirtualThreads() {
       return useVirtualThreads;
+   }
+
+   public static boolean isVirtualThreadsEnabled() {
+      return INSTANCE.isVirtualThreadsEnabled();
    }
 
    public static Thread createThread(ThreadGroup threadGroup, Runnable target, boolean useVirtualThread) {
@@ -65,6 +69,11 @@ public class ThreadCreator {
 
       @Override
       public boolean isVirtual(Thread thread) {
+         return false;
+      }
+
+      @Override
+      public boolean isVirtualThreadsEnabled() {
          return false;
       }
    }

--- a/commons/jdk21/src/main/java/org/infinispan/commons/jdk21/ThreadCreatorImpl.java
+++ b/commons/jdk21/src/main/java/org/infinispan/commons/jdk21/ThreadCreatorImpl.java
@@ -28,4 +28,9 @@ public class ThreadCreatorImpl implements org.infinispan.commons.spi.ThreadCreat
       return thread.isVirtual();
    }
 
+   @Override
+   public boolean isVirtualThreadsEnabled() {
+      return true;
+   }
+
 }

--- a/commons/spi/src/main/java/org/infinispan/commons/spi/ThreadCreator.java
+++ b/commons/spi/src/main/java/org/infinispan/commons/spi/ThreadCreator.java
@@ -12,4 +12,6 @@ public interface ThreadCreator {
    Optional<ExecutorService> newVirtualThreadPerTaskExecutor();
 
    boolean isVirtual(Thread thread);
+
+   boolean isVirtualThreadsEnabled();
 }

--- a/core/src/test/java/org/infinispan/jmx/ClusteredCacheManagerMBeanTest.java
+++ b/core/src/test/java/org/infinispan/jmx/ClusteredCacheManagerMBeanTest.java
@@ -114,9 +114,9 @@ public class ClusteredCacheManagerMBeanTest extends MultipleCacheManagersTest {
 
       objectName = getCacheManagerObjectName(JMX_DOMAIN, "DefaultCacheManager", NON_BLOCKING_EXECUTOR);
       assertTrue(server.isRegistered(objectName));
-      assertEquals(ThreadCreator.useVirtualThreads() ? -1L : 30000L,
+      assertEquals(ThreadCreator.isVirtualThreadsEnabled() ? -1L : 30000L,
             server.getAttribute(objectName, "KeepAliveTime"));
-      assertEquals(ThreadCreator.useVirtualThreads() ? -1 : TestCacheManagerFactory.NAMED_EXECUTORS_THREADS_NO_QUEUE,
+      assertEquals(ThreadCreator.isVirtualThreadsEnabled() ? -1 : TestCacheManagerFactory.NAMED_EXECUTORS_THREADS_NO_QUEUE,
             server.getAttribute(objectName, "MaximumPoolSize"));
    }
 }

--- a/server/runtime/src/test/java/org/infinispan/server/task/ServerTaskEngineBlockingTest.java
+++ b/server/runtime/src/test/java/org/infinispan/server/task/ServerTaskEngineBlockingTest.java
@@ -40,7 +40,7 @@ public class ServerTaskEngineBlockingTest {
 
                @Override
                public Void call() {
-                  if (ThreadCreator.useVirtualThreads()) {
+                  if (ThreadCreator.isVirtualThreadsEnabled()) {
                      assertThat(Thread.currentThread())
                            .withFailMessage(String.format("%s is not a virtual thread", Thread.currentThread()))
                            .matches(ThreadCreator::isVirtual);

